A96-retry-otel-stats.md

```diff
@@ -0,0 +1,79 @@
+## A96: OTel Metrics for Retries
+
+*   Author: Yash Tibrewal (@yashykt)
+*   Approver: Mark Roth (@markdroth)
+*   Status: In Review
+*   Implemented in:
+*   Last updated: 2025-04-02
+*   Discussion at: https://groups.google.com/g/grpc-io/c/bFUHkcBA9cw
+
+## Abstract
+
+Propose OpenTelemetry metrics for gRPC retries.
+
+## Background
+
+Since OpenCensus has been sunsetted in favor of OpenTelemetry, gRPC has been
+developing its own OpenTelemetry plugin that is meant to replace the OpenCensus
+plugin ([A66]). This document proposes the OpenTelemetry version of the retry
+metrics originally proposed in [A45].
+
+### Related Proposals:
+
+*   [A45]: Exposing OpenCensus Metrics and Tracing for gRPC retry
+*   [A66]: OpenTelemetry Metrics
+*   [A79]: Non-per-call Metrics Architecture
+
+[A45]: A45-retry-stats.md
+[A66]: A66-otel-stats.md
+[A79]: A79-non-per-call-metrics-architecture.md
+
+## Proposal
+
+Metric Name                          | Type      | Unit                | Labels                                         | Description
+------------------------------------ | --------- | ------------------- | ---------------------------------------------- | -----------
+grpc.client.call.retries             | Histogram | {retry}             | grpc.method (required), grpc.target (required) | Number of retry or hedging attempts excluding transparent retries made during the client call. The original attempt is not counted as a retry/hedging attempt.
```

Comment:
We'll need to define the buckets for the histograms, right?

Reply:
added

Reply:
Exponential Histograms are just a type of histograms. Our APIs do not need anything more to support exponential histograms. The OpenTelemetry SDK user can simply provide the buckets to use when creating the histogram view.

Aggregation for histograms, which in most cases is calculating percentiles (50% / 95% / 99 %) is always approximate since we no longer have the exact values, but just the bucket ranges.